flask_dance/contrib/authentiq.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import unicode_literals
+
+from functools import partial
+
+from flask.globals import LocalProxy, _lookup_app_object
+
+from flask_dance.consumer import OAuth2ConsumerBlueprint
+
+try:
+    from flask import _app_ctx_stack as stack
+except ImportError:
+    from flask import _request_ctx_stack as stack
+
+
+__maintainer__ = "Pieter Ennes <support@authentiq.com>"
+
+
+def make_authentiq_blueprint(
+        client_id=None, client_secret=None, scope=None, redirect_url=None,
+        redirect_to=None, login_url=None, authorized_url=None,
+        session_class=None, backend=None, hostname="connect.authentiq.io"):
+    """
+    Make a blueprint for authenticating with authentiq using OAuth 2. This requires
+    a client ID and client secret from authentiq. You should either pass them to
+    this constructor, or make sure that your Flask application config defines
+    them, using the variables AUTHENTIQ_OAUTH_CLIENT_ID and
+    AUTHENTIQ_OAUTH_CLIENT_SECRET.
+
+    Args:
+        client_id (str): The client ID for your application on Authentiq.
+        client_secret (str): The client secret for your application on Authentiq.
+        scope (str, optional): comma-separated list of scopes for the OAuth token.
+        redirect_url (str): the URL to redirect to after the authentication
+            dance is complete.
+        redirect_to (str): if ``redirect_url`` is not defined, the name of the
+            view to redirect to after the authentication dance is complete.
+            The actual URL will be determined by :func:`flask.url_for`.
+        login_url (str, optional): the URL path for the ``login`` view.
+            Defaults to ``/authentiq``.
+        authorized_url (str, optional): the URL path for the ``authorized`` view.
+            Defaults to ``/authentiq/authorized``.
+        session_class (class, optional): The class to use for creating a
+            Requests session. Defaults to
+            :class:`~flask_dance.consumer.requests.OAuth2Session`.
+        backend: A storage backend class, or an instance of a storage
+                backend class, to use for this blueprint. Defaults to
+                :class:`~flask_dance.consumer.backend.session.SessionBackend`.
+        hostname (str, optional): If using a private instance of authentiq CE/EE,
+            specify the hostname, default is ``connect.authentiq.io``
+
+    :rtype: :class:`~flask_dance.consumer.OAuth2ConsumerBlueprint`
+    :returns: A :ref:`blueprint <flask:blueprints>` to attach to your Flask app.
+    """
+    scope = set(["openid"] + (scope or "").split())
+    scope = " ".join(scope)
```

Comment:
I think you're trying to be too fancy here. I recommend that you just set `scope="openid"` in the function definition, so that "openid" is the default value. Let the developer decide what the scope should be, and pass that value through without modification.